common/presenters/table/object-to-table-head.js

```diff
@@ -1,17 +1,49 @@
 const { pickBy } = require('lodash')
+const queryString = require('query-string')
 
 const i18n = require('../../../config/i18n')
 
-function objectToTableHead({ head } = {}) {
-  if (!head) {
-    return undefined
+function buildUrl(head) {
+  const { query, key } = head
+  return queryString.stringify({
+    ...query,
+    sortBy: key,
+    sortDirection: (query && query.sortDirection) === 'asc' ? 'desc' : 'asc',
+  })
+}
+
+function buildWrapper(head) {
+  const label = i18n.t(head.html)
+  const url = buildUrl(head)
+  if (head.sortable) {
+    return `<a class="sortable-table__button" role="button" href="?${url}">${label}</a>`
   }
+  return label
+}
 
-  return pickBy({
-    ...head,
-    html: head.html ? i18n.t(head.html) : undefined,
-    text: head.text ? i18n.t(head.text) : undefined,
-  })
+function objectToTableHead(query) {
+  return function objectToTableHead({ head } = {}) {
+    if (!head) {
+      return undefined
+    }
+    if (head.sortable) {
+      head.query = query
```

Comment:
This feels a little heavy and difficult to read to set the query on each head element. You have access to the function that you call that builds the element and URL so maybe just pass it in once rather than having to attach it to the head object:

```javascript
html: head.html ? buildWrapper(head, query) : undefined,
```

instead of:

```javascript
html: head.html ? buildWrapper(head) : undefined,
```